plugins/triggerer.go

```diff
@@ -0,0 +1,209 @@
+package plugins
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"github.com/alexandre-normand/slackscot"
+	"github.com/alexandre-normand/slackscot/store"
+	"regexp"
+	"sort"
+	"strings"
+	"text/tabwriter"
+)
+
+// Triggerer holds the plugin data for the triggerer plugin
+// The triggerer plugin consists of a command to register a new trigger with its answer along
+// with a hear action that will listen and react if one of the registered triggers is said
+type Triggerer struct {
+	slackscot.Plugin
+	triggerStorer store.StringStorer
+}
+
+const (
+	// triggererPluginName holds identifying name for the triggerer plugin
+	triggererPluginName = "triggerer"
+)
+
+var registerTriggerRegex = regexp.MustCompile("(?i)trigger on (.+) with (.+)")
+var deleteTriggerRegex = regexp.MustCompile("(?i)forget trigger on (.+)")
```

Comment:
Why aren't `registerTriggerRegex` and `deleteTriggerRegex` `const`s?

Reply:
Oh you and your good questions. Only basic types `int`, `string`, etc. can be proper constants in `go`.

Reply:
Thank goodness we don't have any modern useful features like enforcement of immutability.

Reply:
I'd like to say that once a `regex` is created, it can't be mutated but it's only _almost_ true because there's this one that can change the `regex` behavior: https://golang.org/pkg/regexp/#Regexp.Longest.

Reply:
Also, 😬